src/transformers/models/gpt_neo/modeling_gpt_neo.py

```diff
@@ -134,156 +134,70 @@ def load_tf_weights_in_gpt_neo(model, config, gpt_neo_checkpoint_path):
-
-        if is_key_value:
-            padded_tensor = padded_tensor.transpose(-2, -1)
-        return padded_tensor
-
-    @staticmethod
-    def _split_seq_length_dim_to(tensors, dim_factor_1, dim_factor_2):
-        """
-        Splits sequence length dim of tensors into `dim_factor_1` and `dim_factor_2` dims
-        """
-        batch_size = tensors.shape[0]
-        split_dim_shape = (batch_size, dim_factor_1, dim_factor_2)
-
-        if len(tensors.shape) == 3:
-            return torch.reshape(tensors, split_dim_shape + (-1,))
-        elif len(tensors.shape) == 2:
-            return torch.reshape(tensors, split_dim_shape)
-        else:
-            raise ValueError(f"Input vector rank should be one of [2, 3], but is: {len(tensors.shape)}")
-
-    @staticmethod
-    def create_local_attention_mask(batch_size, seq_length, window_size, device, attention_mask=None):
-        block_length, num_blocks = GPTNeoAttentionMixin._get_block_length_and_num_blocks(seq_length, window_size)
-        indices = torch.arange(seq_length, dtype=torch.long, device=device).repeat(batch_size, 1)
-
-        query_indices = GPTNeoAttentionMixin._split_seq_length_dim_to(indices, num_blocks, block_length)
-        key_indices = GPTNeoAttentionMixin._look_back(indices, block_length, window_size, is_key_value=False)
-
-        # create mask tensor such that each block contains a causal_mask for that block
-        causal_mask = torch.ge(query_indices.unsqueeze(-1), key_indices.unsqueeze(-2))
-
-        if attention_mask is None:
-            attention_mask = torch.ones(batch_size, seq_length, dtype=torch.long, device=device)
+class GPTNeoSelfAttention(nn.Module):
+    def __init__(self, config, attention_type):
+        super().__init__()
 
-        # A block can also be padded because of the _look_back operation
-        # look back into the attention_block such that it will also get padded the same way
-        # and have 0s in the padded position
-        attention_mask = GPTNeoAttentionMixin._look_back(attention_mask, block_length, window_size, is_key_value=False)
-        attention_mask = attention_mask.unsqueeze(-2)  # Add an extra dimension to account for hidden_dim
+        max_positions = config.max_position_embeddings
+        bias = torch.tril(torch.ones((max_positions, max_positions), dtype=torch.uint8)).view(
+            1, 1, max_positions, max_positions
+        )
+        if attention_type == "local":
+            bias = bias ^ torch.tril(bias, -config.window_size)
 
-        # Multiply the causal_mask with attention_mask so the padded positions (by _look_back operation)
-        # will contain 0s.
-        # This also makes sure that other positions ignored by the attention_mask will also be ignored
-        # in the causal_mask.
-        causal_mask = causal_mask * attention_mask
+        self.register_buffer("bias", bias)
+        self.register_buffer("masked_bias", torch.tensor(-1e9))
```

Comment:
This will save to the state dict - is that voluntary? If not, you can add the `persistent=False` flag (introduced in pytorch 1.6)